src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadataWorker.cs

```diff
@@ -402,26 +469,27 @@ private static void CopyFromCachedResult(BuildInfo buildInfo, IEnumerable<string
             Logger.Log(LogLevel.Info, $"'{inputs.ToDelimitedString()}' keep up-to-date since '{buildInfo.TriggeredUtcTime.ToString()}', cached result from '{buildInfo.OutputFolder}' is used.");
             relativeFiles.Select(s => Path.Combine(outputFolderSource, s)).CopyFilesToFolder(outputFolderSource, outputFolder, true, s => Logger.Log(LogLevel.Info, s), null);
         }
-        
-        private static Task<MetadataItem> GetProjectMetadataFromCacheAsync(Project project, string outputFolder, ProjectDocumentCache documentCache, bool forceRebuild, bool preserveRawInlineComments, string filterConfigFile)
+
+        private static Task<MetadataItem> GetProjectMetadataFromCacheAsync(Project project, Compilation compilation, string outputFolder, ProjectDocumentCache documentCache, bool forceRebuild, bool preserveRawInlineComments, string filterConfigFile, IEnumerable<IMethodSymbol> extensionMethods)
         {
             var projectFilePath = project.FilePath;
             var k = documentCache.GetDocuments(projectFilePath);
             return GetMetadataFromProjectLevelCacheAsync(
                 project,
                 new[] { projectFilePath, filterConfigFile },
                 s => Task.FromResult(forceRebuild || s.AreFilesModified(k.Concat(new string[] { filterConfigFile }))),
-                s => project.GetCompilationAsync(),
+                s => Task.FromResult(compilation),
```

Comment:
nice catch! this way, seems extensionmethods have to be a complete search

---
In reply to: [65648768](https://github.com/dotnet/docfx/pull/339#discussion_r65648768) [](ancestors = 65648768)